app/controllers/api/v1/groups_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# Controller for managing groups in the API version 1 namespace.
+class Api::V1::GroupsController < ApplicationController
+  before_action :set_group, only: %i[show update destroy]
+
+  def index
+    @groups = Group.all
```

Comment:
Is this endpoint really required? Could you think through any realistic scenario where we may have to return all available groups to the client? If there are no such usecases, please do not expose this endpoint